### PR TITLE
Migrate from usage of channels to waitgroups on app/analytics.go

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Migrate from usage of channels to waitgroups on app/analytics.go

#### Ticket Link
Not yet

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
Same comment as in https://github.com/mattermost/mattermost-server/pull/16926#issuecomment-778951060.

**Comment by jespino** [comment]
@agnivade I'll do this changes after the other PR gets merged (to avoid the new-dependency code addition).

**Comment by jespino** [comment]
@agnivade done, now looks much cleaner :)

File: app/analytics.go [hunk 1]
```diff
@@ -244,96 +227,80 @@ func (a *App) GetAnalytics(name string, teamID string) (model.AnalyticsRows, *mo
...
-		sessionChan := make(chan store.StoreResult, 1)
-		go func() {
-			count, err2 := a.Srv().Store.Session().AnalyticsSessionCount()
-			sessionChan <- store.StoreResult{Data: count, NErr: err2}
-			close(sessionChan)
-		}()
-
-		var fileChan chan store.StoreResult
-		var hashtagChan chan store.StoreResult
+		var g errgroup.Group
```
Comment (agnivade) [inline app/analytics.go:230]: I'd suggest naming this to `g2` or `gInternal` to avoid confusion.

**Review by agnivade — COMMENTED** [review]
Looks great!

File: app/analytics.go [hunk 2]
```diff
@@ -43,115 +44,106 @@ func (a *App) GetAnalytics(name string, teamID string) (model.AnalyticsRows, *mo
...
+				return model.NewAppError("GetAnalytics", "app.channel.analytics_type_count.app_error", nil, err.Error(), http.StatusInternalServerError)
+			}
+			return nil
+		})
+
+		var privateChannelsCount int64
+		g.Go(func() error {
+			var err error
+			privateChannelsCount, err = a.Srv().Store.Channel().AnalyticsTypeCount(teamID, model.CHANNEL_PRIVATE)
+			if err != nil {
```
Comment (mgdelacroix) [inline app/analytics.go:61]: nit for consistency

```suggestion
			if privateChannelsCount, err = a.Srv().Store.Channel().AnalyticsTypeCount(teamID, model.CHANNEL_PRIVATE); err != nil {
```

**Review by mgdelacroix — APPROVED** [review]

File: app/analytics.go (hunk 1 above)
Comment (jespino) [reply app/analytics.go]: done

**Review by agnivade — APPROVED** [review]

**Comment by agnivade** [comment]
Can be qa-deferred.

**Pull request merged**